### PR TITLE
Switch the steps of post-analysis and publish artifact. Convert sarif file extension to csv

### PR DESCRIPTION
**Pull request by sima-zhu**
The PR is to have the right order for post-analysis and publish step. 
Also, add csv step into published artifacts, so people can check the errors easily.

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=855108&view=logs&s=96ac2280-8cb4-5df5-99de-dd2da759617d

### CODE REVIEW

File: eng/pipelines/aggregate-reports.yml [hunk 1]
```diff
@@ -126,6 +123,30 @@ jobs:
...
+        $results | ForEach-Object {
+          $csvRows += [pscustomobject]@{
+            ruleId = $_.ruleId
+            ruleIndex = $_.ruleIndex
+            fingerprints = $_.fingerprints
+            message = $_.message
+            properties = $_.properties
+            rank = $_.rank
+            fileName = $_.analysisTarget.uri
+            startLine = $_..locations.physicalLocation.region.startLine
```
Comment (weshaggard) [inline eng/pipelines/aggregate-reports.yml:138]: Is this correct? Would it be 2 dots?

File: eng/pipelines/aggregate-reports.yml [hunk 2]
```diff
@@ -126,6 +123,30 @@ jobs:
         # GdnBreakOutputBaseline: baseline
       condition: succeededOrFailed()
       continueOnError: true
+    - pwsh: |
+        $results = (Get-Content $(Pipeline.Workspace)\.gdn\.r\CredScan\001\CredScan-matches.sarif | ConvertFrom-Json).runs.results
```
Comment (weshaggard) [inline eng/pipelines/aggregate-reports.yml:127]: Is this consistent for all languages? If so we may want to consider sharing this script.

File: eng/pipelines/aggregate-reports.yml (hunk 1 above)
Comment (sima-zhu) [reply eng/pipelines/aggregate-reports.yml]: Typo. changed

File: eng/pipelines/aggregate-reports.yml (hunk 2 above)
Comment (sima-zhu) [reply eng/pipelines/aggregate-reports.yml:127]: Right, it is consistent for all languages.

File: eng/common/pipelines/templates/steps/convert-credScan-sarif-csv.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+parameters:
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/convert-credScan-sarif-csv.yml:1]: As we talked for now just put this template under the eng/pipelines/templates/steps and not under eng/common and then please file an issue in the tools repo to move this to eng/common and assign it to me.

**Review by weshaggard — APPROVED** [review]
Move the template and finish your testing but otherwise this looks reasonable.

**Pull request merged**